component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1279,16 +1268,39 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
         String subUrl = smsProviderUrl.getProtocol();
         if (subUrl.equals(SMSOTPConstants.HTTPS)) {
             httpConnection = (HttpsURLConnection) smsProviderUrl.openConnection();
-            connection = getConnection(httpConnection, context, headerString, payload, httpResponse,
-                    receivedMobileNumber, smsMessage, otpToken, httpMethod);
         } else {
             httpConnection = (HttpURLConnection) smsProviderUrl.openConnection();
-            connection = getConnection(httpConnection, context, headerString, payload, httpResponse,
-                    receivedMobileNumber, smsMessage, otpToken, httpMethod);
         }
+        connection = getConnection(httpConnection, context, headerString, payload, httpResponse, mobile,
+                smsMessage, otpToken, httpMethod);
         return connection;
     }
 
+    /**
+     * Get the corresponding encoded value based on the provided content-type.
+     *
+     * @param contentType   The content type.
+     * @param value         String that needed to be encoded.
+     * @return The encoded value.
+     * @throws IOException
```

Comment:
When we are adding the method comment, we need to make sure we add all the descriptions.

Reply:
Addressed with [5186befd6f64157a75c30fcd83ce4e4473e65485](https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/124/commits/5186befd6f64157a75c30fcd83ce4e4473e65485)